added support for arbitrary encoding/unicode

cgadgil opened the pull request:
Added support to process "arbitrary" encoding in text. Parametrized the char set rather than depend on hard coded ascii.

spro commented:
Looks good, thank you. If you could clean up some of the commented out lines I'll merge it in.

By the way this would be great to have in https://github.com/spro/char-rnn.pytorch

cgadgil commented:
Sure. I have cleaned up a bit more and added some params. Will submit new
request after testing a bit more.


On Sat, Jul 22, 2017 at 11:50 AM, Sean Robertson <notifications@github.com>
wrote:

> Looks good, thank you. If you could clean up some of the commented out
> lines I'll merge it in.
>
> By the way this would be great to have in https://github.com/spro/char-
> rnn.pytorch
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/spro/practical-pytorch/pull/44#issuecomment-317202902>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAOXzo297N4oAFZjqGWZRNeFwJ3o6G1Qks5sQkSCgaJpZM4OZMCm>
> .
>

hershaw commented on char-rnn-generation/helpers.py:
```diff
@@ -10,19 +10,25 @@
-    return file, len(file)
+    #global all_characters
+    #global n_characters
+    s = open(filename).read()
```
While you're working on this, mind fixing up the read to do the following:

```
with open(filename) as fh:
    ...
```
So that the file handle gets closed after the function returns. Thanks!
